core/src/main/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfo.scala

```diff
@@ -57,6 +58,16 @@ sealed abstract class UTXOSpendingInfo {
   def redeemScriptOpt: Option[ScriptPubKey]
 
   def scriptWitnessOpt: Option[ScriptWitness]
+
+  def conditionalPath: ConditionalPath
+}
+
+sealed trait ConditionalPath
+
+object ConditionalPath {
+  case object NoConditionsLeft extends ConditionalPath
```

Comment:
nit: maybe just call this `None`?

Reply:
So long-term (i.e. in the next PR) this will represent the leaf of a condition binary tree which is why I went with the name, I also think it makes more clear what kind of error message should be written if you are expecting more conditions (say deep in some tree).

Your call though, still think I should change it? (I don't have a strong opinion on this one)